src/librustc/hir/lowering.rs

```diff
@@ -1095,15 +1095,15 @@ impl<'a> LoweringContext<'a> {
                         impl_trait_node_id,
                         DefPathData::ImplTrait,
                         ExpnId::root(),
-                        DUMMY_SP
+                        c.span,
```

Comment:
Preexisting.... But, when reading this I felt it would have been easier to understand (at least the diff) if this was `constraint.span` or `assoc_ty_constraint.span` instead of a one letter name. Mind fixing also?